lib/_stream_readable.js

```diff
@@ -499,8 +499,12 @@ Readable.prototype.read = function(n) {
     // If the length is currently zero, then we *need* a readable event.
     if (state.length === 0)
       state.needReadable = true;
-    // Call internal read method
-    this._read(state.highWaterMark);
+    try {
+      // Call internal read method
+      this._read(state.highWaterMark);
+    } catch (err) {
+      errorOrDestroy(this, err);
```

Comment:
Should we return after this? I don't think any part of the rest of the logic should be executed.